### ISSUE TITLE
Video append of XXXXXX failed for segment #X in playlist XX-placeholder-uri-XX

### ISSUE THREAD

**Issue by efb4f5ff-1298-471a-8973-3d47447115dc** [FreeTubeApp/FreeTube#761]
**Behavior of the program**
When opening a video to watch i get this error message in the player and the video is not playable. 
This happens randoms to me.

**Expected behavior**
Dont get this message and play this vid.

**To Reproduce**
Please add all steps to reproduce the behavior:
1. Go to 'watch a vid
2. See error

**Screenshots**
![Screenshot (26)](https://user-images.githubusercontent.com/73130443/97900557-b3f9e080-1d32-11eb-989f-6dea03e264dc.png)

**Desktop (please complete the following information):**
 - OS: [Win 10 Home]
 - OS Version: [1909]
 - FreeTube version: [0.9.2.]
 - FreeTube build: [Release]

**Additional context**

**Comment by GilgusMaximus** [comment]
As far as I figured it out, this is related to a problem with the player and the information in the dash formats.
For me it only happened when I switched to 2k / 4k resolutions tho.

Do you have such a high resolution screen or switched to such a resolution?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> 
> 
> As far as I figured it out, this is related to a problem with the player and the information in the dash formats.
> For me it only happened when I switched to 2k / 4k resolutions tho.
> 
> Do you have such a high resolution screen or switched to such a resolution?

No i do not have such high screen resolution or swtitched to such resolution

**Comment by rootedetc** [comment]
For me this started appearing today on build 0.10.0-nightly-411.
<img width="1213" alt="Screenshot 2021-01-03 at 11 06 57" src="https://user-images.githubusercontent.com/61841347/103476191-d8664d80-4db3-11eb-88f1-1242131c05cd.png">

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Can confirm that this is still the case for me also on 411

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I'm really getting crazy of this error. 

https://user-images.githubusercontent.com/73130443/104901702-a210fc80-5975-11eb-9eba-815d6f0a00f3.mp4

Things i tried to resolve the error, but didnt work.
- Closing the application -> Reopen -> Check if it's gone.
- Removing it from watch history -> Check if it's gone.
- Removing it from watch history -> Closing the application -> Reopen -> Check if it's gone.

**Comment by pawlosck** [comment]
I confirm, that this bug exists in latest git version. I tried to open this movie https://www.youtube.com/watch?v=LXb3EKWsInQ

**Comment by ShadowOfGannicus** [comment]
Same issue on some videos, but only when using "Dash Formats" video format. Found that switching default video formats to "legacy formats" in settings seems to be viable workaround for that issue.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Just wanted to give a quick update: I cant seem to reproduce this bug anymore. I tried out all videos related to this issue.

**Comment by pawlosck** [comment]
I confirm. I can't see this bug since few weeks.

**Comment by GilgusMaximus** [comment]
Good to hear, then I am going to close this

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@GilgusMaximus could u reopen this i just encounterd this issue again. v0.13.1-nightly-930 Beta

![Capture](https://user-images.githubusercontent.com/73130443/122647008-b10cfe00-d111-11eb-9050-3d9f9632734e.PNG)

**Comment by GilgusMaximus** [comment]
Sure. Still a weird issue tho

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@ChunkyProgrammer do u have any ideas on for this?

**Comment by ChunkyProgrammer** [comment]
I think it's an upstream issue https://github.com/videojs/video.js/issues/7008#issuecomment-848634147

**Comment by k8188219** [comment]
@ChunkyProgrammer that work around is for older browser. latest chromium already fix that bug.
If you want to investigate this problem. I suggest you try [videojs/mux.js](https://github.com/videojs/mux.js).
Modify the example code to load the video which cause this error and add the following line.
`sourceBuffer.timestampOffset = -2;`

**Comment by KingBright** [comment]
> @ChunkyProgrammer that work around is for older browser. latest chromium already fix that bug.
> If you want to investigate this problem. I suggest you try [videojs/mux.js](https://github.com/videojs/mux.js).
> Modify the example code to load the video which cause this error and add the following line.
> `sourceBuffer.timestampOffset = -2;`

This exactly can fix my problem, but how can I fix it with videojs.js?

**Comment by k8188219** [comment]
@KingBright 
I don't know what problem you are dealing with.
You need to provide more detailed information about your problem and how you fix it.
Maybe consider opening another issue.

**Comment by EvilGremlin** [comment]
What i discoverd: 
Due to networks overload i have to often toggle VPN (by changing my LAN IP to be exact, actual route change and VPN is handled by my router) to get vidos playing at non-dialup speed, or playing at all. Looks like this error happens when player is waiting for response and buffer is too small, partial (like only audio) or no buffer. Cannot replicate while there is decent buffer present and video loads fast on both routes.

**Comment by peepopoggers** [comment]
Had this problem playing music as well for some reason
CTRL+R fixed this

**Comment by hl2guide** [comment]
Yep I'm getting: video append of 895921b failed for segment #3 in playlist 10-placeholder-uri-10
when setting to 1080p

**Comment by peepopoggers** [comment]
That is an issue I get for around 1 in 30 videos (when using invidious API). It has never happened when visiting the invidious instance I am using directly, so it is only an issue with FreeTube. I wanted to try invidious again when using FreeTube, however it still has this problem. When I get this error, it is persistent with videos until maybe hours later?? I have not tested but a refresh does not fix.

**Comment by LynHyper** [comment]
If anyone's looking for a fix, do the following if this is happening:
1. Changes quality to desired quality setting.
2. See video fail with usual errors reported above.
3. Change the video format to Legacy and then back to Dash.
4. Try playing again and see that the error doesn't occur this time, if not, repeat.

**Comment by billyray098** [comment]
Facing the same issue when I DOWNGRADE the video quality, haven't tested for upgrading yet but this is getting annoying

**Comment by peepopoggers** [comment]
Solution for now is change to legacy player

**Comment by PikachuEXE** [comment]
@peepopoggers 
You mean legacy video format?
![image](https://user-images.githubusercontent.com/1018543/144954785-ad519ac2-eb05-44df-be11-e320b4c67409.png)

**Comment by peepopoggers** [comment]
yes

**Comment by absidue** [comment]
I haven't tried this yet, but I noticed FreeTube uses video.js 7.14.3 (released on the 27th of July 2021), while the current stable version of video.js on npm is 7.17.0 (released on the 10th of November 2021). It might be worth trying to upgrade the dependencies and seeing if that helps.

**Comment by absidue** [comment]
Okay so upgrading `video.js` to 7.17.0 seems to fix the issue, however it's definitely not a drop in replacement, here is a list of issues i've noticed after upgrading `video.js`, `videojs-overlay` and `videojs-vtt-thumbnails-freetube` to the latest versions (the other video.js plugins are all up to date already):

* quality selector is missing from the player (i'm guessing `videojs-http-source-selector` needs an update, although it seems to be abandoned, last commit was in 2019)
* when clicking on a video it doesn't automatically start playing
* shorts overflow the bottom of the window and take up the whole width of the player, instead of being restricted to the visible area like before
* there are probably other issues too that I haven't noticed yet.

**Comment by peepopoggers** [comment]
interesting investigation. this error is one of the most annoying to it would be great to get this fixed.

**Comment by absidue** [comment]
After my investigation I started trying to fix the issues, after spending way to much time on it and only making it worse, I'm leaving it up to you guys that know and understand vue.js, video.js and the black magic that is FreeTube codebase better than I do.

**Comment by nsde** [comment]
I keep getting this error 
![image](https://user-images.githubusercontent.com/67185896/159267942-14eb75b7-b632-4c27-a1b3-0ca927cdba52.png)

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
![I keep getting this error](https://user-images.githubusercontent.com/46166740/160417802-90b0366d-0568-476e-93ca-deb1760b079b.png)

**Comment by VictoriousFan** [comment]
I'm getting this same error after clicking on a random video in my subscriptions
I'm using the latest FlatPak distrubution, with no config changes at all I've only subscribed to a few channels. Pop!_OS

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
v0.16.0-beta 
video append of 0b failed for segment `#123` in playlist 4-placeholder-uri-4

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@4-FLOSS-Free-Libre-Open-Source-Software just upvote the issue we know this issue this persists and your screenshot doesn't provide any information that isn't already provided.

**Comment by nsde** [comment]
> @nsde there hasn't been a FreeTube release since the last time you commented
Sorry, I didn't check that first.
But is there any fork of FreeTube which is maintained actively?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@nsde FreeTube is being actively maintained. See https://github.com/FreeTubeApp/FreeTube/commits/development

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
I tried it with the new version "7.18.1", it did not fix this issue. Still manual CTRL+R required for refresh to resume video.

**Comment by stemy2** [comment]
It happens to me when i'm watching a video in 720p and switch it to 1080p.

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
> Okay so upgrading `video.js` to 7.17.0 seems to fix the issue


Then you can try new [freetube-0.17.0-RC](https://github.com/FreeTubeApp/FreeTube/suites/6899334782/artifacts/267714626) with     ["video.js": "7.18.1",](https://github.com/FreeTubeApp/FreeTube/blob/v0.17.0-RC/package.json#L68)

**Comment by absidue** [comment]
@pawlosck you don't need to build from source if you want to use the 0.17.0 RC, just download it from this link https://github.com/FreeTubeApp/FreeTube/actions/runs/2484662180 (you need to be logged into GitHub)

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
in todays [Release v0.17.0 Beta](https://github.com/FreeTubeApp/FreeTube/releases/tag/v0.17.0-beta) the issue is still not fixed.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@4-FLOSS-Free-Libre-Open-Source-Software there was never anything being mentioned in the release notes that this is fixed.

Please do not clutter the comment section with your comments because they dont contribute anything to solving this issue.

I have also saw this kind of behaviour of u in other issues as well. E.g. only posting a screenshot in the comment section of an error message that is already well explained in the description of the issue. 

Or saying maybe X could work even though it isnt related at all to solving the issue.

Again do not do this!

**Comment by kubo6472** [comment]
Per #2443 which now no longer happens, but this exact thing happens instead.

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
This issue is not stale

**Comment by billyray098** [comment]
> This issue is not stale

yes, experiencing the same problem in the latest version at well, although not as frequently.

**Comment by ChunkyProgrammer** [comment]
Might be relevant https://github.com/iv-org/invidious/issues/2848

**Comment by anonaddict** [comment]
Can confirm this still happens

**Comment by torriem** [comment]
Still happening.  In fact nearly every video will eventually freeze for me unless I switch over to legacy formats.

**Comment by absidue** [comment]
Can anyone who is using a **recent nightly build** still reproduce this? I'm curious if any of the changes we've done since the last stable release have fixed this or whether more are needed.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Closing, I have not encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds). Issue seems to be resolved.
